### ISSUE TITLE
[api-extractor] Cannot assign isExternal

### ISSUE THREAD

**Issue by npirotte** [Microsoft/web-build-tools#1128]
api-extractor version: 7.0.17

Hi, 

I am running into an error whenever I try to export a react component as part of my public api.

```
 throw new node_core_library_1.InternalError(`Cannot assign isExternal=${options.isExternal} for`
            ^

Error: Internal Error: Cannot assign isExternal=true for the symbol React because it was previously registered with isExternal=false
```

I have no idea how to deal with that, this was working fine with version 6. Could anyone give me a hint?

**Comment by npirotte** [comment]
First found: It looks like it's working when I turn esModuleInterop to false (and remove all esModuleInterop type imports)

**Comment by npirotte** [comment]
Second found: It's also working with esModuleInterop, but not if I mix 2 flavors.

And by the way the esModuleInterop output something strange:
`import { default } from "react"`

**Comment by octogonz** [comment]
Can you share a repro branch? We're very close to releasing AE7, so this week I'm going through and retesting various projects to find any remailing blocking bugs.

I wonder if your case is an import form that is not recognized. Yesterday I found this one:

https://github.com/Microsoft/web-build-tools/pull/1127

**Comment by npirotte** [comment]
Let me quickly create something (current case in on a private repo)

**Comment by npirotte** [comment]
https://github.com/collibra/api-extractor-example

But I was not able to reproduce the failure, maybe a combination on small things on my project, I clean up some stuffs and continue investigations.

You can still checkout the `default` issue here https://github.com/collibra/api-extractor-example/blob/master/temp/api-extractor-example.api.ts and here https://github.com/collibra/api-extractor-example/blob/master/dist/typings/typings.d.ts

maybe it's expected behavior, but it makes it hard for reviewer :)

**Comment by octogonz** [comment]
Agreed we should probably emit a better name than `default`.  But your code is mixing together two different ways of importing React:

```ts
import React from "react";
```

```ts
import * as React from "react";
```

That's not a best practice.

BTW it is nontrivial for API Extractor to figure out these two forms refer to the same thing, so even if we fix the `default` name, you may still see two names like `React` and `React_2`.  (Long term we'll fix that as well, but it's a cosmetic issue and thus lower priority.)

**Comment by lizozom** [comment]
I'm now running into this error as well: 
`UnhandledPromiseRejectionWarning: Error: Internal Error: Cannot assign isExternal=true for the symbol React because it was previously registered with isExternal=false`

I tried replacing the few places that were using `import * as React from 'react'` with `import React from 'react'`, but the same error still happens.

How can I find our which file is triggering that error? I tried setting a breakpoint in apps/api-extractor/src/analyzer/AstSymbolTable.ts but couldn't make sense of the options object.

**Comment by nazarhussain** [comment]
Bump into same issue for different object in my case. 

```
Error: [vite:dts] Internal Error: Cannot assign isExternal=false for the symbol HashObject because it was previously registered with isExternal=true
```
